### PR TITLE
Throw error for unused imports

### PR DESCRIPTION
**Pull request by hahnbeelee**
## Summary
As title says.
Specifically implements the [eslint-plugin-unused-imports](https://www.npmjs.com/package/eslint-plugin-unused-imports)

## Test Plan
Style check passes

### CODE REVIEW

File: client/.eslintrc.json [hunk 1]
```diff
@@ -1,5 +1,14 @@
 {
   "parser": "@typescript-eslint/parser",
-  "plugins": ["@typescript-eslint"],
-  "extends": ["next/core-web-vitals", "plugin:@typescript-eslint/recommended", "prettier"]
+  "plugins": ["@typescript-eslint", "unused-imports"],
+  "extends": ["next/core-web-vitals", "plugin:@typescript-eslint/recommended", "prettier"],
+  "rules": {
```
Comment (tobias-mintlify) [inline client/.eslintrc.json:5]: the indentation in this looks wrong

Comment (hahnbeelee) [reply client/.eslintrc.json:5]: this what i mean that we need prettier for json files :blob-glare:

Comment (hahnbeelee) [reply client/.eslintrc.json:5]: I can look into implementing that too

Comment (hahnbeelee) [reply client/.eslintrc.json:5]: It looks fine to me tho :looking-sus:
image

File: client/.eslintrc.json [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+		"@typescript-eslint/no-unused-vars": "off",
+		"unused-imports/no-unused-imports": "error",
+		"unused-imports/no-unused-vars": [
+			"warn",
+			{ "vars": "all", "varsIgnorePattern": "^_", "args": "after-used", "argsIgnorePattern": "^_" }
+		]
+	}
```
Comment (hahnbeelee) [inline client/.eslintrc.json:12]: This setup is recommended in the plugin's readme.
I eventually want `unused-imports/no-unused-vars` to error, but I tried that and there was too much rearchitecting to be done.  sadge

File: client/.eslintrc.json (hunk 1 above)
Comment (hahnbeelee) [reply client/.eslintrc.json:5]: oh jk the indentation was slightly off. Fixed.

**Review by tobias-mintlify — APPROVED** [review]

**Pull request merged**